File: api/src/main/java/de/hipp/pnp/api/fivee/interfaces/FiveEDataProducer.java
package de.hipp.pnp.api.fivee.interfaces;

import de.hipp.pnp.api.fivee.LanguageValue;

import java.util.Map;

public interface FiveEDataProducer {

    // <Language, <LanguageKey, LanguageEntry>>
    public Map<String, Map<String, Map<String, Map<String, LanguageValue>>>> getAllLanguageKeys();

    // <LanguageKey, LanguageEntry>
    public Map<String, Map<String, Map<String, Map<String, LanguageValue>>>> getLanguageKeysByGameTypeAndLanguage(int gameType, String locale);
}

File: api/src/main/kotlin/de/hipp/pnp/api/fivee/interfaces/FiveEDataProducer.kt
package de.hipp.pnp.api.fivee.interfaces

import de.hipp.pnp.api.locale.BookLocale

interface FiveEDataProducer {
    fun getAllLanguageKeys(): MutableMap<String, BookLocale>?

    fun getLanguageKeysByGameTypeAndLanguage(gameType: Int, locale: String?): MutableMap<String, BookLocale>?
}
